sdk/core/core-client-lro-rest/src/getLongRunningHelper.ts

```diff
@@ -0,0 +1,183 @@
+      // which is an opaque URL provided by caller, the service sends this in one of the following headers: operation-location, azure-asyncoperation or location
+      // depending on the lro pattern that the service implements. If non is provided we default to the initial path.
+      const response = (await client
+        .pathUnchecked(path ?? initialResponse.request.url)
+        .get()) as TResult;
+
+      // Translating the rest response into a response that the LRO engine understands
+      const lroResponse = getLroResponse(response);
+
+      // Hand off the information we have to the LRO engine to continue polling
+      return createGetLROState<TResult>(
+        poller,
+        config,
+        finalStateVia
+      )(lroResponse.rawResponse, lroResponse.flatResponse);
+    },
+    retrieveAzureAsyncResource: async (path) => {
+      // This is the function executed when a final request is needed to get the result once the operation completed
+      // we use the path provided by the engine when this callback is executed or default to the initial path.
+      const response: HttpResponse = await client
+        .pathUnchecked(path ?? initialResponse.request.url)
+        .get();
+
+      const lroResponse = getLroResponse(response);
+
+      // retreiveAzureAsyncResource is called as a final request to get the result
+      // we need to set the done property as true since there are no additional requests to do
+      lroResponse.done = true;
+
+      return createGetLROState<TResult>(poller, {})(
+        lroResponse.rawResponse,
+        // flatResponse is used when there is additional deserialization
+        // since RLC doesn't use mappers flatResponse is the same as rawResponse
+        lroResponse.flatResponse as TResult
+      );
+    },
+  };
+
+  return new LROPoller(pollerOptions, poller);
+}
+
+/**
+ * Converts a Rest Client response to a response that the LRO engine knows about
+ * @param response - a rest client http response
+ * @returns - An LRO response that the LRO engine can work with
+ */
+function getLroResponse<TResult extends HttpResponse>(response: TResult): LROState<TResult> {
+  if (Number.isNaN(response.status)) {
+    throw new TypeError(`Status code of the response is not a number. Value: ${response.status}`);
+  }
+
+  return {
+    flatResponse: response,
+    rawResponse: {
+      ...response,
+      statusCode: Number.parseInt(response.status),
+      body: response.body as LROBody,
+    },
+    done: false,
+  };
+}
+
+function isFinalStateVia(param?: LROPollerOptions | FinalStateVia): param is FinalStateVia {
+  if (
+    typeof param === "string" &&
+    ["azure-async-operation", "location", "original-uri"].includes(param)
+  ) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function isLROPollerOptions(param?: LROPollerOptions | FinalStateVia): param is LROPollerOptions {
+  if (typeof param === "object") {
```

Comment:
NIT: Similar feedback as above, can we remove the if?

Reply:
`isLROPollerOptions` has been removed